src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+      const labelContainer = screen.getByTestId("label-container");
+      const validationIcon = screen.getByTestId(`icon-${validationProp}`);
+      expect(labelContainer).toContainElement(validationIcon);
+    });
+
+    it("overrides the tooltip position on the validation icon when the tooltipPosition prop is set and validationOnLabel is true", () => {
+      const useFloatingSpy = jest.spyOn(floatingUi, "useFloating");
+
+      render(
+        <Textarea
+          {...{
+            label: "Label",
+            [validationProp]: "Message",
+            tooltipPosition: "bottom",
+            validationOnLabel: true,
+          }}
+        />
+      );
+
+      expect(useFloatingSpy).toHaveBeenCalledWith(
+        expect.objectContaining({ placement: "bottom" })
+      );
+
+      useFloatingSpy.mockRestore();
+    });
+  }
+);
+
+test.each(validationTypes)(
+  "when %s validation prop is true boolean, does not render any validation iconn",
+  (validationProp) => {
+    render(<Textarea {...{ [validationProp]: true }} />);
+    expect(
+      screen.queryByTestId(`icon-${validationProp}`)
+    ).not.toBeInTheDocument();
+  }
+);
+
+test("should set the aria-label on the Help component to the value of the helpAriaLabel prop", () => {
+  render(<Textarea label="foo" labelHelp="bar" helpAriaLabel="baz" />);
+
+  expect(screen.getByRole("button")).toHaveAttribute("aria-label", "baz");
+});
+
+test("should set the Help component's text content to the value of the labelHelp prop", () => {
+  render(<Textarea label="foo" labelHelp="bar" helpAriaLabel="baz" />);
+
+  screen.getByRole("button").focus();
+  expect(screen.getByTestId("tooltip")).toHaveTextContent("bar");
+});
+
+test.each(["info", "warning", "error"])(
+  "with id prop provided, %s prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element",
+  (validationType) => {
+    render(<Textarea id="foo" label="bar" {...{ [validationType]: "test" }} />);
+    const textarea = screen.getByRole("textbox");
+    textarea.focus();
+
+    expect(textarea).toHaveAttribute("aria-describedby", "foo-validation");
+  }
+);
+
+test.each(["info", "warning", "error"])(
+  "with id prop not provided, %s prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element",
+  (validationType) => {
+    render(<Textarea label="bar" {...{ [validationType]: "test" }} />);
+    const textarea = screen.getByRole("textbox");
+    textarea.focus();
+
+    expect(textarea).toHaveAttribute(
+      "aria-describedby",
+      `${mockedGuid}-validation`
+    );
+  }
+);
+
+test("renders the hint when inputHint prop is provided", () => {
+  render(<Textarea inputHint="foo" />);
+  expect(screen.getByText("foo")).toBeInTheDocument();
+});
+
+test("assigns the input hint a guid as id and references it in the aria-describedby of the input", () => {
```

Comment:
**suggestion(non-breaking)**: for the accessible name and description tests, what do you think about refining the test descriptions to focus on the intended outcomes rather than the implementation details? This can help make the purpose of each test clearer.

```tsx
// test("assigns the input hint a guid as id and references it in the aria-describedby of the input", () => {
test("textarea uses input hint text as its accessible description when inputHint prop is provided", () => {

// test("when id and fieldHelp are both present, the id of the field help is added to 'aria-describedby' in the input", () => {
test("textarea uses field help content as its accessible description when fieldHelp prop is provided", () => {

// test("with id prop provided, error prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element", () => {
test("adds validation tooltip content to textarea's accessible description, when error prop is a string and textarea is focused", () => {
```

Reply:
I agree - if I change the test implementation in the way @edleeks87 has suggested, then this is a better way to word the test description.